providers/default.rb

```diff
@@ -224,11 +231,40 @@ def move_opt_chef(src, dest)
   end
 end
 
+def prepare_windows
+  copy_opt_chef(chef_install_dir, chef_backup_dir)
+  Kernel.spawn('c:/windows/system32/schtasks.exe /F /RU SYSTEM /create /sc minute /mo 1 /tn Chef_upgrade /tr "powershell.exe c:/opscode/chef_upgrade.ps1"')
+  FileUtils.rm_rf "#{chef_install_dir}/bin/chef-client.bat"
+end
+
 def execute_install_script(install_script)
   if windows?
     powershell_script 'name' do
       code <<-EOH
-    #{install_script}
+        $command = {
+          Get-Service chef-client -ErrorAction SilentlyContinue | stop-service
+
+          if ((Get-WmiObject Win32_Process -Filter "name = 'ruby.exe'" | Select-Object CommandLine | select-string 'opscode').count -gt 0) { exit 8 }
+
+          Remove-Item "#{chef_install_dir}" -Recurse -Force
+
+          if (test-path "#{chef_install_dir}") { exit 3 }
+
+          #{install_script}
+
+          Remove-Item "c:/opscode/chef_upgrade.ps1"
+          c:/windows/system32/schtasks.exe /delete /f /tn Chef_upgrade
+
+          Get-Service chef-client -ErrorAction SilentlyContinue | Start-service
+          c:/opscode/chef/bin/chef-client.bat
```

Comment:
We shouldn't need to hardcode this location if we have #{chef_install_dir} ...?

Reply:
Valid point. Will update this after @smurawski has revisited this PR.